### PR TITLE
[1118] create static preference data

### PR DESCRIPTION
**Pull request by misaka**
### Context

https://trello.com/c/em4QA1AI/1118-create-static-preferences-data-for-ucas-api-testing

UCAS would like some static data to show up for their preferences so they can test importing them.

The values are defined in the [API docs](https://github.com/DFE-Digital/manage-courses-backend/blob/master/docs/api.md#entity-documentation-4):

| Parameter              | Data type          | Possible values                                                                                            | Description                                                                                          |
| ---------------------- | ------------------ | ---------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------- |
| utt_application_alerts | Text               | `No, not required`, `Yes, required`, `Yes - only my programmes` or `Yes - for accredited programmes only`  | New UTT Application alerts                                                                           |
| type_of_gt12           | Text               | `Coming / Enrol`, `Coming or Not`, `No response` or `Not coming`                                           |                                                                                                      |

### Changes proposed in this pull request

Add these attributes to the serializer for providers and generate their value randomly, but consistent according to the provider code.

### CODE REVIEW

**Comment by misaka** [comment]
[Spec API for UCAS contact details](https://trello.com/c/Flx0PnGG/1069-spec-api-for-ucas-contact-details)

File: app/serializers/provider_serializer.rb [hunk 1]
```diff
@@ -69,4 +70,41 @@ def postcode
...
+      'No response',
+      'Not coming',
+    ]
+
+    select_value_for_provider(@object.provider_code, values)
+  end
+
+private
+
+  def select_value_for_provider(provider_code, values)
```
Comment (heidar) [inline app/serializers/provider_serializer.rb]: If we're just grabbing a random value, would it be better to call this `select_random_value_for_provider` to be explicit about that?

Comment (misaka) [reply app/serializers/provider_serializer.rb]: I agree naming could be better, but this is temporary anyway, so will go away (soon?!).

File: app/serializers/provider_serializer.rb [hunk 2]
```diff
@@ -69,4 +70,41 @@ def postcode
...
+    # Using `to_i(36)` is an easy, cheap way to convert 'A1' into a consistent
+    # hash. ex. 'A1'.to_i(36) == 361
+    index = provider_code.to_i(36) % values.count
+    values[index]
```
[the remark this answers is not in the document]
Comment (timabell) [reply app/serializers/provider_serializer.rb]: that would make the data unstable

[the remark this answers is not in the document]
Comment (misaka) [reply app/serializers/provider_serializer.rb]: wut he sez

**Pull request merged**